prow/cmd/deck/tide.go

```diff
@@ -155,60 +159,103 @@ func (ta *tideAgent) updateHistory() error {
 	return nil
 }
 
-func (ta *tideAgent) filterHiddenPools(pools []tide.Pool) []tide.Pool {
-	if len(ta.hiddenRepos()) == 0 {
-		return pools
-	}
+func (ta *tideAgent) matchingIDs(ids []string) bool {
+	return len(ids) > 0 && len(ta.tenantIDs) > 0 && sets.String{}.Insert(ta.tenantIDs...).HasAll(ids...)
+}
 
+func (ta *tideAgent) filterHiddenPools(pools []tide.Pool) []tide.Pool {
 	filtered := make([]tide.Pool, 0, len(pools))
 	for _, pool := range pools {
-		needsHide := matches(pool.Org+"/"+pool.Repo, ta.hiddenRepos())
-		if needsHide && ta.showHidden {
-			filtered = append(filtered, pool)
-		} else if needsHide == ta.hiddenOnly {
+		// curIDs are the IDs associated with all PJs in the Pool
+		// We want to add the ID associated with the OrgRepo for extra protection
+		curIDs := sets.String{}.Insert(pool.TenantIDs...)
+		orgRepoID := ta.cfg.GetProwJobDefault(pool.Org+"/"+pool.Repo, "*").TenantID
+		// If OrgRepo is associated with no TenantID, ignore it.
+		if orgRepoID != "" && orgRepoID != config.DefaultTenantID {
+			curIDs.Insert(orgRepoID)
+		}
+		if ta.matchingIDs(curIDs.List()) {
+			// Deck has tenantIDs and they match with the pool
 			filtered = append(filtered, pool)
+		} else if len(ta.tenantIDs) == 0 {
```

Comment:
can we save one layer of indentation by:
```
else if needsHide := matches(pool.Org+"/"+pool.Repo, ta.hiddenRepos()); needsHide {
```
contingent on the previous `if` condition uses `len(ta.tenantIDs)`

Reply:
We would still need to double check that the len(tenantIDs ==0) b/c the prev if statement would get skipped if len(tenantIDs) >0 but they did not match.

Reply:
that's true. I think it would work better by:
```
if len(ta.tenantIDs) > 0 {
        if  ta.matchingIDs(curIDs.List()) {
			// Deck has tenantIDs and they match with the pool
			filtered = append(filtered, pool)
        }
} else if needsHide := matches(pool.Org+"/"+pool.Repo, ta.hiddenRepos()); needsHide {
```

Reply:
Couldn't exactly do this. Wasn't able to save an indenting butI think it should be easier to follow now.